src/internal/pfsdb/commits.go

```diff
@@ -1206,11 +1206,11 @@ func pickCommitAncestorOf(ctx context.Context, ancestorOf *pfs.CommitPicker_Ance
 	offset := 0
 	commitPtr := startCommit.ID
 	if err := forEachCommitAncestorUntilRoot(ctx, tx, startCommit.ID, func(parentId, _ CommitID) error {
-		commitPtr = parentId
-		offset++
 		if uint32(offset) == ancestorOf.Offset {
 			return errutil.ErrBreak
 		}
+		commitPtr = parentId
```

Comment:
Changes to this file fix a bug I found with the recursive navigation. That ended up being important for traversing down to the root.